### ISSUE TITLE
build failure with upcoming clang-16

### ISSUE THREAD

**Issue by orlitzky** [monitoring-plugins/monitoring-plugins#1789]
clang-15 tried to enable the following by default:

* `-Werror=implicit-function-declaration`
* `-Werror=implicit-int`
* `-Werror=strict-prototypes`

This caused some breakage, so they're delaying it until clang-16. But it's still coming. With those flags, monitoring-plugins fails to build:

```
/usr/include/openssl/macros.h:62:52: note: expanded from macro 'OSSL_DEPRECATED'
#     define OSSL_DEPRECATED(since) __attribute__((deprecated))
                                                   ^
sslutils.c:198:9: error: call to undeclared function
      'np_net_ssl_check_certificate'; ISO C99 and later do not support implicit
      function declarations [-Werror,-Wimplicit-function-declaration]
```

(This was reported to us at https://bugs.gentoo.org/871519 on Gentoo)

**Comment by RincewindsHat** [comment]
Ich changed the title, since the problem occurs with the openssl bindings and `check_disk` does not use those. But this impacts several other plugins.

Thank you for reporting this issue.

**Comment by RincewindsHat** [comment]
The error should be fixed with c4704e163ebf54277ff901f06f09126ef3a3bc7f when it is merged. The OpenSSL API deprecation is probably a topic for another time

**Comment by RincewindsHat** [comment]
The rest should be fixed with #1848

**Comment by RincewindsHat** [comment]
@orlitzky Parts of this should be fixed by #1868
I tried the master with `Debian clang version 16.0.6 (15)` and it compiles (with warnings).
I will close this issue, please reopen it, if you think this is not yet resolved.

**Comment by orlitzky** [comment]
Wow, that wound up being a lot of work. Thank you!
